src/EoS/TaubMathews/CPU_EoS_TaubMathews.cpp

```diff
@@ -0,0 +1,336 @@
+#ifndef __CUDACC__
+void EoS_SetAuxArray_TaubMathews( double AuxArray_Flt[], int AuxArray_Int[] )
+{
+   AuxArray_Flt[0] = ( OPT__UNIT ) ? MOLECULAR_WEIGHT * Const_amu / Const_kB * (UNIT_E/UNIT_M)
+
+                                   : MOLECULAR_WEIGHT;
+   AuxArray_Flt[1] = 1.0 / AuxArray_Flt[0];
+} // FUNCTION : EoS_SetAuxArray_TaubMathews
+#endif // #ifndef __CUDACC__
+
+
+
+// =============================================
+// II. Implement EoS conversion functions
+//     (1) EoS_HTilde2Temp_*
+//     (2) EoS_Temp2HTilde_*
+//     (3) EoS_Temper2CSqr_*
+//     (4) EoS_GuessHTilde_*
+// =============================================
+
+
+
+//-------------------------------------------------------------------------------------------------------
+// Function    :  EoS_GuessHTilde_TaubMathews
+// Description :  Guess a reduced enthalpy for Newton-Raphson iteration
+// Note        :
+// Parameter   :  Con         : Conservative variables
+//                Constant    : The constant on the left side of Eq. A3 in "Tseng et al. 2021, MNRAS, 504, 3298"
+//                AuxArray_*  : Auxiliary arrays (see the Note above)
+//                Table       : EoS tables
+// Return      :  GuessHTilde : guessed reduced enthalpy
+//-------------------------------------------------------------------------------------------------------
+GPU_DEVICE_NOINLINE
+static real EoS_GuessHTilde_TaubMathews( const real Con[], real* const Constant, const double AuxArray_Flt[],
+                                         const int AuxArray_Int[], const real *const Table[EOS_NTABLE_MAX] )
+{
+   real GuessHTilde, Discrimination;
+
+   real Msqr = SQR(Con[1]) + SQR(Con[2]) + SQR(Con[3]);
+   real Dsqr = SQR(Con[0]);
+   real abc = (real)1.0 / Dsqr;
+   real E_D = Con[4] / Con[0];
+   real M_Dsqr = abc * Msqr;
+   real M_D = SQRT( M_Dsqr );
+
+   // note that replacing a^2-b^2 with (a+b)*(a-b) helps alleviate a catastrophic cancellation
+   real X = SQRT( E_D*E_D + (real)2.0*E_D );
+   real Y = X + M_D;
+   real Z = X - M_D;
+   *Constant = Y * Z;
+
+#  ifdef GAMER_DEBUG
+   if ( *Constant <= (real)TINY_NUMBER )
+      printf( "ERROR : f(HTilde) = %14.7e <= %13.7e (D %13.7e, E_D %13.7e, M_D %13.7e) in %s !!\n",
+              *Constant, TINY_NUMBER, Con[0], E_D, M_D, __FUNCTION__ );
+#  endif
+
+   real A = (real)437.0 * M_Dsqr + (real)117.0;
+   real B = (real)1.0 + M_Dsqr;
+
+   // Eq. A7 in "Tseng et al. 2021, MNRAS, 504, 3298"
+   Discrimination  = (real)3240000.0 * SQR( B );
+   Discrimination /= SQR( A );
+
+   if ( *Constant >= Discrimination )
+   {
+      // Eq. A6 in "Tseng et al. 2021, MNRAS, 504, 3298"
+      GuessHTilde  = (real)4./3. * SQRT( *Constant );
+   }
+   else
+   {
+      // Eq. A4 in "Tseng et al. 2021, MNRAS, 504, 3298"
+      real C       = (real)43.0*M_Dsqr + (real)63.0;
+      real F       = (real)75.0*B;
+      real G       = (real)125.0*B*C*(*Constant);  
+      GuessHTilde  = SQRT(G);
```

Comment:
I think this line should be replaced with `GuessHTilde  = G;`. @ChunYen-Chen Can you cross-check it?